src/EntityComponentManager.cc

```diff
@@ -2314,3 +2314,22 @@ void EntityComponentManager::ResetTo(const EntityComponentManager &_other)
   tmpCopy.ApplyEntityDiff(*this, ecmDiff);
   this->CopyFrom(tmpCopy);
 }
+
+/////////////////////////////////////////////////
+std::optional<Entity> EntityComponentManager::EntityByName(
+    const std::string &_name) const
+{
+  std::optional<Entity> entity;
+  this->Each<components::Name>([&](const Entity _entity,
+        const components::Name *_entityName)->bool
+  {
+    if (_entityName->Data() == _name)
+    {
+      entity = _entity;
+      return false;
+    }
+    return true;
+  });
```

Comment:
I think this can be simplified to:

```suggestion
   Entity entByName = EntityByComponents(components::Name(_name));
  if (entByName != kNullEntity)
    entity = entByName;
```